extensions/wikia/Track/Track.php

```diff
@@ -1,59 +1,130 @@
+		global $wgTitle, $wgContLanguageCode, $wgDBname, $wgUser, $wgCityId, $wgGAUserIdSalt;
+
+		$skinName = RequestContext::getMain()->getSkin()->getSkinName();
+		$adContext = ( new AdEngine2ContextService() )->getContext( $wgTitle, $skinName );
+		$hubFactory = WikiFactoryHub::getInstance();
+
+		$params = [
+			'v' => static::GA_VERSION,
+			'ds' => 'backend',
+			't' => $type,
+			'ni' => 1,
+			'dl' => $wgTitle->getFullURL(),
+			'ul' => $wgContLanguageCode,
+			'de' => 'UTF-8',
+			'dt' => $wgTitle->getText(),
+			'ec' => $category,
+			'ea' => $action,
+			'el' => $label,
+			'_utma' => $_COOKIE[ '__utma' ] ?? '',
+			'_utmz' => $_COOKIE[ '__utmz' ] ?? '',
+			'cid' => $_COOKIE[ '_ga' ] ? explode( ".", $_COOKIE[ '_ga' ] )[ 2 ] : uniqid(),
+			'tid' => $tid,
+			// custom dimensions
+			'cd1' => $wgDBname,
+			'cd2' => $wgContLanguageCode,
+			'cd3' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd4' => $skinName,
+			'cd5' => $wgUser->isAnon() ? 'anon' : 'user',
+			'cd8' => WikiaPageType::getPageType(),
+			'cd9' => $wgCityId,
+			'cd13' => AdTargeting::getEsrbRating(),
+			'cd14' => isset( $adContext[ 'opts' ][ 'showAds' ] ) ? 'Yes' : 'No',
+			'cd15' => WikiaPageType::isCorporatePage(),
+			'cd17' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd18' => implode( ',', $hubFactory->getWikiCategories( $wgCityId ) ),
+			'cd19' => WikiaPageType::getArticleType(),
+			'cd21' => $wgTitle->getArticleID(),
+			'cd23' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_FREQUENT ) ? 'Yes' : 'No',
+			'cd24' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_LIFETIME ) ? 'Yes' : 'No',
+			'cd25' => $wgTitle->getNamespace(),
+		];
+		if ( !$wgUser->isAnon() ) {
+			$params[ 'uid' ] = md5( $wgUser->getId() . $wgGAUserIdSalt );
+		}
+		$params = array_merge( $params, $extraParams );
+		$url = static::GA_URL . '/debug/collect?' .
+			   implode( $for_html ? '&amp;' : '&', array_map( function ( $k, $v ) {
+				   return "{$k}={$v}";
+			   }, array_keys( $params ), $params ) );
+
+		return $url;
+	}
+
+	private static function getGATrackingIds() {
+		global $wgDevelEnvironment, $wgIsGASpecialWiki, $wgUser;
+
+		$tids = [ ];
+		$tids[] = $wgDevelEnvironment ? 'UA-32129070-2' : 'UA-32129070-1';
+		if ( $wgIsGASpecialWiki ) {
+			$tids[] = $wgDevelEnvironment ? 'UA-32132943-2' : 'UA-32132943-1';
+		}
+		if ( !$wgUser->isAnon() ) {
+			$tids[] = $wgDevelEnvironment ? 'UA-32132943-8' : 'UA-32132943-7';
+		}
+
+		return $tids;
+	}
+
+	private static function getViewJS( $param = null ) {
 		global $wgDevelEnvironment;
 
 		// Fake beacon and varnishTime values for development environment
 		if ( !empty( $wgDevelEnvironment ) ) {
 			$script = '<script>var beacon_id = "ThisIsFake", varnishTime = "' . date( "r" ) . '";</script>';
 
 		} else {
-			$url = Track::getURL('view', '', $param);
+			$url = Track::getURL( 'view', '', $param );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$url" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)